src/vnsw/opencontrail-vrouter-netns/opencontrail_vrouter_netns/vrouter_hyperv.py

```diff
@@ -0,0 +1,469 @@
+import os
+import time
+import sys
+import argparse
+import subprocess
+import uuid
+
+import netaddr
+import requests
+import json
+import paramiko
+
+
+def validate_uuid(val):
+    try:
+        if str(uuid.UUID(val)) == val:
+            return val
+    except (TypeError, ValueError, AttributeError):
+        raise ValueError('Invalid UUID format')
+
+
+def powershell(cmds):
+    cmds = ["powershell.exe", "-NonInteractive"] + cmds
+    try:
+        output = subprocess.check_output(cmds, shell=True)
+    except subprocess.CalledProcessError as e:
+        raise
+    return output
+
+
+class HyperVManager(object):
+    SNAT_RT_TABLES_ID = 42
+    NAME_LEN = 14
+    WINGW_PREFIX = 'contrail-wingw-'
+    LEFT_DEV_PREFIX = 'eth'
+    RIGHT_DEV_PREFIX = 'eth'
+    PORT_TYPE = 'NameSpacePort' # should maybe be NovaVMPort?
+    BASE_URL = "http://localhost:9091/port"
+    HEADERS = {'content-type': 'application/json'}
+
+    HYPERV_GENERATION = '2'
+    RAM_GB = '1GB'
+
+    USERNAME = 'ubuntu'
+    PASSWORD = 'ubuntu'
+
+    INJECT_IP_SCRIPT_REL_PATH = "vrouter_hyperv_inject_ip.ps1"
+    WAIT_FOR_VM_TIME_SEC = 10
+
+    HOST_MGMT_IP = "169.254.150.1"
+    MGMT_PREFIX_LEN = 16
+    MGMT_IP_RANGE = netaddr.IPRange("169.254.150.10", "169.254.150.254")
+    MGMT_SUBNET_MASK = "255.255.0.0"
+    MGMT_HOST_ADAPTER_NAME = "mgmt_snat_adapter"
+
+    def __init__(self, vm_uuid, nic_left, nic_right, wingw_vm_name=None,
+                 vm_location=None, vhd_path=None,
+                 mgmt_vswitch_name=None, vrouter_vswitch_name=None,
+                 gw_ip=None):
+
+        self.vm_uuid = vm_uuid
+        self.nic_left = nic_left
+        self.nic_right = nic_right
+        self.gw_ip = gw_ip
+
+        self.vm_location = vm_location
+        self.vhd_path = vhd_path
+        self.mgmt_vswitch_name = mgmt_vswitch_name
+        self.vrouter_vswitch_name = vrouter_vswitch_name
+
+        self.wingw_name = self.WINGW_PREFIX + self.vm_uuid \
+            if wingw_vm_name is None else wingw_vm_name
+
+        self.nic_left['name'] = (self.LEFT_DEV_PREFIX + "1")
+                                #self.nic_left['uuid'])[:self.NAME_LEN]
+        self.nic_right['name'] = (self.RIGHT_DEV_PREFIX + "2")
+                                #self.nic_right['uuid'])[:self.NAME_LEN]
```

Comment:
I'm not sure if the name here must be unique (if we require uuids here).

On Linux, these are `int-uuid1` and `gw-uuid2`. Here it's just `eth1` and `eth2`.